engine/configstore/supabaseLegacy.ts

```diff
@@ -238,6 +293,73 @@ export const newSupabaseProviderLegacyDeploy = (
   };
 };
 
+const matchesToMatchMulti = (matches: Flag["data"]["matches"], ns: string) => ({
+  op: "or",
+  matchers: matches.map((m) => ({
+    ...m.props,
+    __resolveType: includeNamespace(m.key, ns).replace("functions", "matchers"),
+  })),
+  __resolveType: "$live/matchers/MatchMulti.ts",
+});
+
+const flagsToConfig = (
+  entrypoint: typeof baseEntrypoint,
+  flags: Pick<Flag, "key" | "data" | "name">[],
+  ns: string,
+) => {
+  return flags.reduce((curr, flag) => {
+    const catchall = curr[catchAllConfig];
+    const pageIds = flag.data.effect?.props?.pageIds;
+    const pageId = Array.isArray(pageIds) ? pageIds[0] as number : undefined;
+    if (!pageId) {
+      return curr;
+    }
+    const page = curr[pageId];
+    if (!page) {
+      return curr;
+    }
+    return {
+      ...curr,
+      [catchAllConfig]: {
+        ...catchall,
+        handler: {
+          ...catchall.handler,
+          flags: [
+            ...catchall.handler.flags,
+            {
+              __resolveType: flag.key,
+            },
+          ],
+        },
+      },
+      [flag.key]: {
+        name: flag.key,
+        routes: {
+          [page.path]: {
+            page: {
+              __resolveType: pageId,
+            },
+            __resolveType: "$live/handlers/fresh.ts",
+          },
+        },
+        matcher: matchesToMatchMulti(flag.data.matches, ns),
+        __resolveType: "$live/flags/audience.ts",
+      },
+    };
+  }, entrypoint);
+};
+const pagesToConfig = (
+  p: Page[],
+  flags: Pick<Flag, "key" | "data" | "name">[],
+  ns: string,
+) => {
+  const configs = p.sort((pageA, pageB) =>
+    pageA.state === "global" ? -1 : pageB.state === "global" ? 1 : 0
```

Comment:
global pages should be mapped first to be available in other components that depends on it